java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -349,23 +335,20 @@ private API getXPAPI(ExpressionTree et) {
       }
       MemberSelectTree mst = (MemberSelectTree) mit.getMethodSelect();
       String methodName = mst.getIdentifier().toString();
-      List<Map<String, Object>> methodPropertiesForMethodName =
-          configMethodProperties.get(methodName);
-      if (methodPropertiesForMethodName != null) {
-        return getXPAPI(mit, methodPropertiesForMethodName);
+      if (!disabled && configMethodProperties.containsKey(methodName)) {
+        return getXPAPI(mit, configMethodProperties.get(methodName));
       }
     }
     return API.UNKNOWN;
   }
 
   private API getXPAPI(
-      MethodInvocationTree mit, List<Map<String, Object>> methodPropertiesForMethodName) {
-    for (Map<String, Object> currentMethodProperties : methodPropertiesForMethodName) {
-      Integer argumentIndex = getArgumentIndexFromMap(currentMethodProperties);
-
+      MethodInvocationTree mit, ImmutableCollection<PiranhaMethodRecord> methodRecordsForName) {
+    for (PiranhaMethodRecord methodRecord : methodRecordsForName) {
       // when argumentIndex is specified, if mit's argument at argIndex doesn't match xpFlagName,
       // skip to next method property map
-      if (argumentIndex != null) {
+      if (methodRecord.getArgumentIdx().isPresent()) {
```

Comment:
Can we store the `methodRecord.getArgumentIdx()` locally and re-use it to avoid verbosity?

Reply:
I don't think it saves too much. We'd be replacing a call to `.getArgumentIdx()` with adding a new `Optional<Integer>` local. Either way we need to check for `isPresent()` and then do `get()` to get the actual value, so this seemed clearer to me. That said, I don't feel strongly either way, can rewrite to introduce the local.

Reply:
I am not referring to the performance (the compiler will likely take care of this) but to the readability. Feel free to update (or not).

Reply:
I am also not talking about performance. I am just not sure it's more readable/less verbose/less idiomatic to go `var.getField.isPresent()` and then `T field = var.getField.get()` vs `Optional<T> maybeField = var.getField()` then `maybeField.isPresent` then `T field = maybeField.get()`.

But I have no strong opinion here, so let me update.